### DISCUSSION TITLE
Failed ：no EAP key found for hosts

### DISCUSSION THREAD

**Discussion by 3157696** [strongswan/strongswan#1927] [Q&A]
HELLO
everyone
I referred to this case to configure eap-mschapv2 authentication but it failed.
https://www.strongswan.org/testing/testresults/ikev2/rw-eap-mschapv2-id-rsa/
What's the problem? what other configuration do I need to provide?

Client is windows 10
Thanks.

swanctl.conf:
`conn-defaults {
        local_addrs=vps.domain.com
        remote_addrs=%any
        version=2
        reauth_time=10800
        unique=replace
        proposals=a-b-c
        local {
                auth=pubkey
                certs=vps.cer
                id=vps.domain.com
                }
        remote {
                }
        }
child-defaults {
                local_ts =0.0.0.0/0
                rekey_time=5400
                updown=/usr/local/libexec/ipsec/_updown iptables
                esp_proposals=a-b-c
                inactivity=1800
                mode=tunnel
        }
#---------------------------------------------------------------------
                        # conn
connections {
#---------------------------------------------------------------------
eap-mschapv2 : conn-defaults {
        remote {
                eap_id=%any
                auth=eap-mschapv2
                }
        children {
        eap:child-defaults {
                                }
                }
        pools=eap_pool_rw
     }
#---------------------------------------------------------------------
                        # secrets && pools
#---------------------------------------------------------------------
secrets {
        eap-huang {
                id = huang
                secret = abc123ABC
        }
}
#---------------------------------------------------------------------
pools{
        eap_pool_rw {
                addrs=192.168.10.0/27
        }
}
#---------------------------------------------------------------------
authorities{
        ca {
        cacert=ca.cer
        }
        }
#---------------------------------------------------------------------
#include conf.d/*.conf

`

log:
#swanctl -T
12[NET] received packet: from a.b.86.255[161] to c.d.182.53[500] (376 bytes)
12[ENC] parsed IKE_SA_INIT request 0 [ SA KE No N(FRAG_SUP) N(NATD_S_IP) N(NATD_D_IP) V V V V ]
12[IKE] received MS NT5 ISAKMPOAKLEY v9 vendor ID
12[IKE] received MS-Negotiation Discovery Capable vendor ID
12[IKE] received Vid-Initial-Contact vendor ID
12[ENC] received unknown vendor ID: 01:52:8b:bb:c0:06:96:12:18:49:ab:9a:1c:5b:2a:51:00:00:00:02
12[IKE] a.b.86.255 is initiating an IKE_SA
12[CFG] selected proposal: IKE:AES_GCM_16_256/PRF_HMAC_SHA2_384/ECP_384
12[IKE] remote host is behind NAT
12[IKE] sending cert request for "CN=Public Root RSACA"
12[ENC] generating IKE_SA_INIT response 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) CERTREQ N(FRAG_SUP) N(CHDLESS_SUP) N(MULT_AUTH) ]
12[NET] sending packet: from c.d.182.53[500] to a.b.86.255[161] (313 bytes)
09[NET] received packet: from a.b.86.255[25948] to c.d.182.53[4500] (572 bytes)
09[ENC] parsed IKE_AUTH request 1 [ EF(1/4) ]
09[ENC] received fragment #1 of 4, waiting for complete IKE message
11[NET] received packet: from a.b.86.255[25948] to c.d.182.53[4500] (572 bytes)
11[ENC] parsed IKE_AUTH request 1 [ EF(2/4) ]
11[ENC] received fragment #2 of 4, waiting for complete IKE message
04[NET] received packet: from a.b.86.255[25948] to c.d.182.53[4500] (332 bytes)
04[ENC] parsed IKE_AUTH request 1 [ EF(4/4) ]
04[ENC] received fragment #4 of 4, waiting for complete IKE message
13[NET] received packet: from a.b.86.255[25948] to c.d.182.53[4500] (572 bytes)
13[ENC] parsed IKE_AUTH request 1 [ EF(3/4) ]
13[ENC] received fragment #3 of 4, reassembled fragmented IKE message (1802 bytes)
13[ENC] parsed IKE_AUTH request 1 [ IDi CERTREQ CPRQ(ADDR DNS NBNS SRV) SA TSi TSr ]
13[IKE] received cert request for "CN=Public Root RSACA"
13[IKE] received 76 cert requests for an unknown ca
13[CFG] looking for peer configs matching c.d.182.53[%any]...a.b.86.255[192.168.2.189]
13[CFG] selected peer config 'eap-mschapv2'
13[IKE] initiating EAP_IDENTITY method (id 0x00)
13[IKE] authentication of 'vps.domain.com' (myself) with RSA signature successful
13[ENC] generating IKE_AUTH response 1 [ IDr CERT AUTH EAP/REQ/ID ]
13[IKE] sending end entity cert "C=US, CN=vps.domain.com"
13[NET] sending packet: from c.d.182.53[4500] to a.b.86.255[25948] (1168 bytes)
10[NET] received packet: from a.b.86.255[25948] to c.d.182.53[4500] (72 bytes)
10[ENC] parsed IKE_AUTH request 2 [ EAP/RES/ID ]
10[IKE] received EAP identity 'huang'
10[IKE] initiating EAP_MSCHAPV2 method (id 0x34)
10[ENC] generating IKE_AUTH response 2 [ EAP/REQ/MSCHAPV2 ]
10[NET] sending packet: from c.d.182.53[4500] to a.b.86.255[25948] (97 bytes)
08[NET] received packet: from a.b.86.255[25948] to c.d.182.53[4500] (136 bytes)
08[ENC] parsed IKE_AUTH request 3 [ EAP/RES/MSCHAPV2 ]
**08[IKE] no EAP key found for hosts 'vps.domain.com' - 'huang'
08[IKE] EAP-MS-CHAPv2 verification failed, retry (1)**
09[MGR] ignoring request with ID 3, already processing
08[ENC] generating IKE_AUTH response 3 [ EAP/REQ/MSCHAPV2 ]
08[NET] sending packet: from c.d.182.53[4500] to a.b.86.255[25948] (114 bytes)
04[NET] received packet: from a.b.86.255[25948] to c.d.182.53[4500] (136 bytes)
04[IKE] received retransmit of request with ID 3, retransmitting response
04[NET] sending packet: from c.d.182.53[4500] to a.b.86.255[25948] (114 bytes)

**Comment by tobiasbrunner** [comment]
Your config is invalid. You have to close the `connections` section with a `}` before starting the `secrets` section.

**Reply by 3157696** [reply]
Sorry
This is part of the clip, the editor missed it.
Actually loading it via swanctl -c is normal

**Reply by tobiasbrunner** [reply]
Not sure what you mean, but check the log for what's actually loaded etc.
